datafusion/physical-expr/src/functions.rs

```diff
@@ -690,6 +690,21 @@ pub fn create_physical_fun(
                     let func = specializer_func(args)?;
                     func(args)
                 }
+                DataType::Binary => {
```

Comment:
I think we can make the coercion more general and not have to add something specific in the physical implementation

In general the coercion from one type to another is handled in https://github.com/apache/arrow-datafusion/blob/b6fe8159fd21ea101b84abc2629dbb484c8ad326/datafusion/expr/src/type_coercion/functions.rs#L192

Is there any chance you can try adding a coercion there for binary -> utf8 which would allow DataFusion to automatically try and cast binary --> UTF8 if users try and call a function or operator that requires a Utf8?